draft-ietf-anima-brski-prm.md

```diff
@@ -2111,44 +2274,86 @@ The pledge MAY provide the dedicated endpoint for the Query Pledge Status operat
 
 ~~~~ cddl
   statustrigger = {
       "version": uint,
-      "created-on": tdate,
       "serial-number": text,
-      "status-type": text
+      "created-on": tdate,
+      "status-type": $status-type
   }
+
+  $status-type /= "bootstrap"
 ~~~~
-{: #stat_req_def title="CDDL for unsigned Status Trigger data (statustrigger)" artwork-align="left" sourcecode-markers=“true”}
+{: #stat_req_def title="CDDL for JSON Status Trigger Data (statustrigger)" artwork-align="left"}
 
-The `version` field is included to permit significant changes to the pledge status artifacts in the future.
+The `version` member is included to permit significant changes to the pledge status artifacts in the future.
 The format and semantics in this document follow the status telemetry definitions of {{!RFC8995}}.
-Hence, the version MUST be set to `1`.
+Hence, the `version` SHALL be set to `1`.
 A pledge (or Registrar-Agent) that receives a version larger than it knows about SHOULD log the contents and alert a human.
 
-The `created-on` field contains a standard date/time string following {{!RFC3339}}.
+The `serial-number` member SHALL contain the product-serial-number corresponding to the X520SerialNumber field of the pledge IDevID certificate;
+it can be correlated with the product-serial-number in the signing certificate contained in the JWS Protected Header of the Pledge Status response artifact.
 
-The `serial-number` field takes the product-serial-number corresponding to the X520SerialNumber field of the IDevID certificate of the pledge.
+The `created-on` member SHALL contain the current date and time at tStatus creation as standard date/time string as defined in {{Section 5.6 of !RFC3339}};
+it can be used as reference time for the corresponding Pledge Status response artifact after correlating via the product-serial-number;
+note that pledges may not have synchronized time to provide the created-on date and time on their own.
 
-The `status-type` value defined for BRSKI-PRM Status Query is `bootstrap`.
-This indicates the pledge to provide current status information regarding the bootstrapping status (voucher processing and enrollment of the pledge into the new domain).
+The `status-type` allows for specifying which status information is to be returned.
+As shown in {{stat_req_def}}, BRSKI-PRM currently only defines the enum value `bootstrap` to provide current status information regarding the bootstrapping status (voucher processing and enrollment of the pledge into a domain).
+Other specifications using this artifact may define further enum values, e.g., to query application-related status.
 
-As the Status Query artifact is defined generic, it may be used by other specifications to request further status information using other status types, e.g., for onboarding to get further information about enrollment of application specific LDevIDs or other parameters.
-This is out of scope for this specification.
+TODO(Create an IANA registry for the status-type enum?)
```

Comment:
Should we include this in the IANA Considerations?
It would make managing extensions much easier.

Reply:
Yes, while we thought on different status query types during bootstrapping, the same may be used for further operational parameter. Good idea!

Reply:
New IANA text in chat, including in IANA Considerations.